spot-client/src/spot-remote/ui/loaders/withRemoteControl.js

```diff
@@ -124,117 +42,17 @@ export class RemoteControlLoader extends AbstractLoader {
-            roomInfo,
-            serverConfig: this.props.remoteControlConfiguration
-        })
-        .catch(error => {
-            // In the wrong password case return back to join code entry.
-            if (error === 'not-authorized' || error === 'unrecoverable-error') {
-                this.props.dispatch(addNotification('error', 'Something went wrong'));
-
-                this._onDisconnect();
-
-                return Promise.reject();
-            }
-
-            return Promise.reject(error);
-        });
-    }
-
-    /**
-     * Callback invoked when {@code remoteControlClient} has an update about the
-     * current state of a Spot-TV.
-     *
-     * @param {Object} data - Details of the Spot-TV's current state.
-     * @private
-     * @returns {void}
-     */
-    _onSpotTVStateChange(data) {
-        const newState = {};
-        const { updatedState } = data;
-
-        Object.keys(updatedState).forEach(key => {
-            if (presenceToStoreAsBoolean.has(key)) {
-                newState[key] = updatedState[key] === 'true';
-            } else if (presenceToStoreAsString.has(key)) {
-                newState[key] = updatedState[key];
-            }
-        });
-
-        this.props.dispatch(setSpotTVState(newState));
-
-        if (updatedState.calendar) {
-            try {
-                const events = JSON.parse(updatedState.calendar);
-
-                this.props.dispatch(setCalendarEvents(events));
-            } catch (error) {
-                logger.error(
-                    'Spot-Remote could not parse calendar events',
-                    { error }
-                );
-            }
-        }
-    }
-
-    /**
-     * Clean up the connection to the remote control service because a disconnect
-     * event occurred which cannot be recovered from.
-     *
-     * @private
-     * @returns {void}
-     */
-    _onDisconnect() {
-        logger.error('Spot-Remote could not connect to remote control service');
-
-        this.props.dispatch(clearSpotTVState());
-
-        this._redirectBackToLogin();
-    }
-
-    /**
-     * Changes routes to the join code entry view.
-     *
-     * @private
-     * @returns {void}
-     */
-    _redirectBackToLogin() {
-        this.props.history.push('/');
+        return Promise.reject('Spot Remote connection must be started with action');
     }
 }
 
-/**
- * Selects parts of the Redux state to pass in with the props of
- * {@code RemoteControlLoader}.
- *
- * @param {Object} state - The Redux state.
- * @private
- * @returns {Object}
- */
-function mapStateToProps(state) {
-    return {
-        roomInfo: getRoomInfo(state),
-        remoteControlConfiguration: getRemoteControlServerConfig(state)
-    };
-}
 
-const ConnectedRemoteControlLoader
-    = withRouter(connect(mapStateToProps)(RemoteControlLoader));
+const ConnectedRemoteControlLoader = withRouter(RemoteControlLoader);
 
 export default generateWrapper(ConnectedRemoteControlLoader);
```

Comment:
Is withRemoteControl still necessary to exist?

Reply:
I was thinking this component could be useful to switch to the spinner while the connection is lost, but probably right now that's not the case, right ?

Reply:
Okay, let's keep it around until we finalize between us what the desired behavior should be there. I suppose it's still useful to keep around to show the loading state while an initial connection is being made.